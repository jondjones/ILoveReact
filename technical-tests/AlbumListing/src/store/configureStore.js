import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import songs from '../state/reducers/songs'

// Create Store.
export default () =>{
    const store = createStore(
        combineReducers({songs}),
        applyMiddleware(thunk)
    );

    return store
};