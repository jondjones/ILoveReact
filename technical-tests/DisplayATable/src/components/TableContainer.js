import React, { useEffect, useState, Fragment } from 'react';
import { useDispatch } from "react-redux";
import fetchVariant from '../state/actions/fetchVariant'
import fetchColumns from '../state/actions/fetchColumns'
import styled from "styled-components"

// used stylised component for simple rendering
const Table = styled.div`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
`

const TR = styled.div`
  padding: 8px 4px;
  border: 1px solid black;
`

function TableContainer(props) {
  const { columns, variant } = props;

  // use hook to store state
  const [clone, setsClone] = useState(props.show);

  // call redux to get data
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchVariant());
    dispatch(fetchColumns());
  }, [dispatch]);

  if (!columns && !variant) {
    return <div>Loading</div>;
  }

  // not sure if i got this right.  I have the data in the correct format
  // can do what ever mapping is needed
  const items = Object.values(variant)?.reduce(function(reducer, key) {
    const header = columns?.columns?.[key];
    if (header) {
      reducer.push(header);
    }
    return reducer;
  }, []);

  return (
    <Fragment>
        <SimpleTable items={items} />
      {clone && 
        <div>
          <SimpleTable items={clone} />
        </div>
      }
      <button onClick={() => setsClone(items)}>Clone</button>
    </Fragment>
  );
}

const SimpleTable = (prop) => {
  return (
    <Table>
      {prop.items && prop.items.map((item, index) => 
        <TR key={index}>
          {item}
        </TR>
        )}
      </Table>
)}

export function getKeyByValue(object, value) {
  return Object.keys(object).find(key => object[key] === value);
}

export default TableContainer;