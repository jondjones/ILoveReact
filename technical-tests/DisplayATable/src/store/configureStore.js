import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import variants from '../state/reducers/variants'
import columns from '../state/reducers/columns'

// Create Store.
export default () =>{
    const store = createStore(
        combineReducers({variants, columns}),
        applyMiddleware(thunk)
    );

    return store
};