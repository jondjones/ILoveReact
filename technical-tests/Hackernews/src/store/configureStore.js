import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import news from '../state/reducers/news'
import filter from '../state/reducers/filter'

// Create Store.
export default () =>{
    const store = createStore(
        combineReducers({news, filter}),
        applyMiddleware(thunk)
    );

    return store
};