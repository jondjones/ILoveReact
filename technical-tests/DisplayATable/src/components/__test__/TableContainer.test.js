import React from 'react';
import { shallow, configure } from 'enzyme';
import TableContainer from '../TableContainer';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

configure({adapter: new Adapter()});

const mockStore = configureStore();
const store = mockStore();

describe('TableContainer', () => {
    it('renders TableContainer', () => {
        const component = shallow(
            <Provider store={store}>
                <TableContainer />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
})
