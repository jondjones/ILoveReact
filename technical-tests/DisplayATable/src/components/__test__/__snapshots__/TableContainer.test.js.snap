// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer renders TableContainer 1`] = `ShallowWrapper {}`;
